### PR TITLE
Removed the rlzs_assoc object from the datastore

### PR DESCRIPTION
**Pull request by micheles**
Another step towards the removal of pickled objects, closes https://github.com/gem/oq-risklib/issues/659.
Tests and demos are running here: https://ci.openquake.org/job/zdevel_oq-risklib/1591

### CODE REVIEW

File: openquake/commonlib/source.py [hunk 1]
```diff
@@ -531,6 +535,18 @@ class CompositionInfo(object):
...
+    @classmethod
+    def fake(cls, gsimlt=None):
+        """
+        :returns:
+            a fake `CompositionInfo` instance with the given gsim logic tree
+            object; if None, builds automatically a fake gsim logic tree
+        """
+        fakeSM = SourceModel(
+            'fake', 1,  'b1', [TrtModel('*', eff_ruptures=1)],
+            gsimlt or logictree.GsimLogicTree.from_('FromFile'), 0, 1)
```
Comment (ptormene) [inline openquake/commonlib/source.py]: Please add a comment about these numbers

File: openquake/qa_tests_data/scenario_damage/case_6/expected/collapse_map-gsimltp_@.xml [hunk 2]
```diff
@@ -10,11 +10,11 @@ xmlns:gml="http://www.opengis.net/gml"
...
             </gml:Point>
-            <cf assetRef="ant_13_S1" mean="4.36515E-09" stdDev="9.08539E-09"/>
-            <cf assetRef="ant_13_S2" mean="7.33214E-20" stdDev="1.55999E-19"/>
-            <cf assetRef="ant_13_S3" mean="7.44031E-33" stdDev="1.61591E-32"/>
+            <cf assetRef="ant_13_S1" mean="4.36512E-09" stdDev="9.08534E-09"/>
+            <cf assetRef="ant_13_S2" mean="7.33203E-20" stdDev="1.55996E-19"/>
+            <cf assetRef="ant_13_S3" mean="7.44008E-33" stdDev="1.61585E-32"/>
             <cf assetRef="ant_13_S4" mean="2.80260E-44" stdDev="8.26766E-44"/>
-            <cf assetRef="ant_13_S5" mean="3.50617E-40" stdDev="1.04146E-39"/>
+            <cf assetRef="ant_13_S5" mean="3.50613E-40" stdDev="1.04144E-39"/>
```
Comment (ptormene) [inline openquake/qa_tests_data/scenario_damage/case_6/expected/collapse_map-gsimltp_@.xml]: Please check numbers for all case_6

File: openquake/qa_tests_data/scenario_risk/case_6a/gmpe_logic_tree.xml [hunk 3]
```diff
@@ -7,13 +7,13 @@
...
-                    <uncertaintyModel>BooreAtkinson2008</uncertaintyModel>
-                    <uncertaintyWeight>0.75</uncertaintyWeight>
+                  <uncertaintyModel>BooreAtkinson2008</uncertaintyModel>
+                  <uncertaintyWeight>0.75</uncertaintyWeight>
                 </logicTreeBranch>
                 
                 <logicTreeBranch branchID="b2">
- 					<uncertaintyModel>ChiouYoungs2008</uncertaintyModel>
-                    <uncertaintyWeight>0.25</uncertaintyWeight>
+		  <uncertaintyModel>ChiouYoungs2008</uncertaintyModel>
```
Comment (ptormene) [inline openquake/qa_tests_data/scenario_risk/case_6a/gmpe_logic_tree.xml]: Please fix the indentation

File: openquake/commonlib/source.py (hunk 1 above)
Comment (micheles) [inline openquake/commonlib/source.py]: Done

File: openquake/qa_tests_data/scenario_damage/case_6/expected/collapse_map-gsimltp_@.xml (hunk 2 above)
Comment (micheles) [inline openquake/qa_tests_data/scenario_damage/case_6/expected/collapse_map-gsimltp_@.xml]: After a lengty investigation I found the reason for the small rounding errors. The new branch is calling `RlzsAssoc.combine` which performs an operation `1. - (1. - 0.) (1 - x)` which in theory should give `x`, like in the old branch, but in practice gives a slightly different number. That's fine
and normal given how floating point work.

**Comment by ptormene** [comment]
LGTM

**Pull request merged**